src/lib.rs

```diff
@@ -1577,18 +1585,25 @@ impl<T: Trait> Module<T> {
         account_id: T::AccountId,
         name: Vec<u8>,
         self_introduction: Vec<u8>,
+        post_footer: Option<Vec<u8>>,
     ) -> dispatch::Result {
         // Ensure user name is valid
         Self::ensure_user_name_is_valid(&name)?;
 
         // Ensure self introduction is valid
         Self::ensure_user_self_introduction_is_valid(&self_introduction)?;
 
+        // Ensure post footer is valid
+        if post_footer.is_some() {
```

Comment:
`if let Some(ref post_footer) = opt_post_footer` allows you to match and get parameter in one step, which is safer, then you do not need to do `unwrap`.

Reply:
accepted.